tests/generator/test_email.py

```diff
@@ -0,0 +1,22 @@
+from unittest import TestCase
+from unittest.mock import MagicMock, mock_open, patch
+
+from flask_seeder.generator import Email
+
+MOCK_CONTENTS = [
+    "domain1.com",
+    "domain2.com"
+]
+
+class TestEmailGenerator(TestCase):
+
+    def setUp(self):
+        self.rnd_mock = MagicMock()
+        self.generator = Email(rnd=self.rnd_mock)
+
+    @patch("flask_seeder.generator.read_resource", return_value=MOCK_CONTENTS)
+    def test_generate_email(self, m_read_resource):
```

Comment:
While this tests that the `rnd` is called, it doesn't test anything the `Email` generator produces (i.e. it doesn't test that an actual email address has been generated). Updating this to assert that `result` contains an email address would be a small but great improvement.
This would probably mean changing the `@patch` to have a `side_effect` rather than `return_value` to return a list of one name and one domain so that you can assert a predictable email address. If you have questions here or need help please reach out.

Now looking back at other tests in this project I totally realize there is plenty of room for improvements... :)

Reply:
Have a look at the new test, I must say I am a new to the pytest world so forgive any best-practice omissions but I think it does the job right now. I had to mock the actual `choice` function and took me a bit to figure that out hah.

Reply:
Don't worry, we all need to start somewhere :) I believe mocking `choice` could be avoided by letting the generator use the real `random` here as `read_reasource` will only return a single name and domain, making the results predictable. The test is much better now and does the job so no need to change it for the sake of this PR.